machine/machine_core/ssh_connection.py

```diff
@@ -69,6 +70,24 @@ def message(self, *args):
         if self.verbose:
             sys.stderr.write(" ".join(args) + '\n')
 
+    def print_console_log(self):
+        """Prints VM's console to stderr"""
+        if not self.console_file:
+            return
+
+        try:
+            f = open(self.console_file)
```

Comment:
This needs to go inside the `with` already. I. e. `with open(...) as f: log = read()`. The processing of `console_log` shouldn't be in the context.